jenkins/node-image/Makefile

```diff
@@ -15,14 +15,39 @@
 IMG = gcr.io/k8s-testimages/kubekins-node
 TAG = $(shell date +v%Y%m%d)-$(shell git describe --tags --always --dirty)
 
+# default to 1.6 image
+VERSION:=1.6
+KUBEKINS_TAG:=1.6-v20161205-ad918bc
+
+# Resolve VERSION
+ifeq ($(VERSION),1.4)
+	KUBEKINS_TAG=1.4-v20161130-8958f82
+endif
+ifeq ($(VERSION),1.5)
+	KUBEKINS_TAG=1.5-v20161205-d664d14
+endif
+
 all: build
 
 build:
-	docker build -t $(IMG):$(TAG) .
-	docker tag $(IMG):$(TAG) $(IMG):latest
-	@echo Built $(IMG):$(TAG) and tagged with latest
+	sed 's/__KUBEKINS_TAG/$(KUBEKINS_TAG)/g' Dockerfile > Dockerfile-$(VERSION)
```

Comment:
This looks like a maintenance nightmare. Can we avoid this kind of magic, surprising behavior?